terracotta/drivers/postgresql_meta_store.py

```diff
@@ -0,0 +1,80 @@
+"""drivers/postgresql_meta_store.py
+
+PostgreSQL-backed metadata driver. Metadata is stored in a PostgreSQL database.
+"""
+
+from typing import Mapping, Sequence
+
+import sqlalchemy as sqla
+from terracotta.drivers.relational_meta_store import RelationalMetaStore
+
+
+class PostgreSQLMetaStore(RelationalMetaStore):
+    """A PostgreSQL-backed metadata driver.
+
+    Stores metadata and paths to raster files in PostgreSQL.
+
+    Requires a running PostgreSQL server.
+
+    The PostgreSQL database consists of 4 different tables:
+
+    - ``terracotta``: Metadata about the database itself.
+    - ``key_names``: Contains two columns holding all available keys and their description.
+    - ``datasets``: Maps key values to physical raster path.
+    - ``metadata``: Contains actual metadata as separate columns. Indexed via key values.
+
+    This driver caches key names.
+    """
+    SQL_DIALECT = 'postgresql'
+    SQL_DRIVER = 'psycopg2'
+    SQL_TIMEOUT_KEY = 'connect_timeout'
+
+    MAX_PRIMARY_KEY_SIZE = 2730 // 4  # Max B-tree index size in bytes
+    DEFAULT_PORT = 5432
+
+    def __init__(self, postgresql_path: str) -> None:
+        """Initialize the PostgreSQLDriver.
+
+        This should not be called directly, use :func:`~terracotta.get_driver` instead.
+
+        Arguments:
+
+            postgresql_path: URL to running PostgreSQL server, in the form
+                ``postgresql://username:password@hostname/database``
+
+        """
+        super().__init__(postgresql_path)
+
+        # raise an exception if database name is invalid
+        if not self.url.database:
+            raise ValueError('database must be specified in PostgreSQL path')
+        if '/' in self.url.database.strip('/'):
+            raise ValueError('invalid database path')
+
+    @classmethod
+    def _normalize_path(cls, path: str) -> str:
+        url = cls._parse_path(path)
+
+        path = f'{url.drivername}://{url.host}:{url.port or cls.DEFAULT_PORT}/{url.database}'
+        path = path.rstrip('/')
+        return path
+
+    def _create_database(self) -> None:
+        engine = sqla.create_engine(
+            self.url.set(database='postgres'),  # `.set()` returns a copy with changed parameters
```

Comment:
Postgres does not allow to connect without connecting to a specific database.
`postgres` is a database that is automatically created when a PostgreSQL cluster is created.
The Postgres docs say that this database "is meant as a default database for users and applications to connect to", so it should be a safe default.

Reply:
That is also what I had in mind.

Perhaps add a `DEFAULT_DB_FOR_MANAGING = 'postgres'` (if you can think of a better name, please use that) class variable, such that if someone somehow have managed to delete/rename their `postgres` database, they still have the option of creating a TC database, by overwriting that class variable

Reply:
With a brief explaining comment, I think that is good